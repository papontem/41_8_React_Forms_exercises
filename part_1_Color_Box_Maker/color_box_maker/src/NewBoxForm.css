/* NewBoxForm.css */

.NewBoxForm-container {
    margin: 5px;
}

.NewBoxForm-title{
    font-size:calc(10px + 1vmin);
}
.NewBoxForm {
    /* Add additional styling for the form if needed */
}

.NewBoxForm-label {
    margin: 5px;
}

.NewBoxForm-input {
    margin: 5px;
}

.NewBoxForm-btn {
    margin: 5px;
    padding: 8px 12px;
    background-color: #007bff;
    color: #fff;
    border: none;
    cursor: pointer;
}

.NewBoxForm-btn:hover {
    background-color: #0056b3;
}
