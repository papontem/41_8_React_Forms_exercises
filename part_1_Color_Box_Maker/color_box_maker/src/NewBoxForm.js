import React, { useState } from "react";
/* 
    Renders a form that allows users to specify the width, height, and background color for a new box.
    When the form is submitted, it creates a new box with the specified properties and clears the input values.
*/
function NewBoxForm({ addBox }) {
	// State for form data
	// width, height, and background color for a new box.
	const [newBoxFormData, setNewBoxFormData] = useState({
		width: "",
		height: "",
		backgroundColor: "",
	});

	// Event handler for form input changes
	const handleChange = (e) => {
		const { name, value } = e.target;
		setNewBoxFormData((prevData) => ({
			...prevData,
			[name]: value,
		}));
	};

	// Event handler for form submission
	const handleSubmit = (e) => {
		e.preventDefault();

		console.log("Form submitted:", newBoxFormData);
		addBox(newBoxFormData);

		// Clear input values after form submission
		setNewBoxFormData({
			width: "",
			height: "",
			backgroundColor: "",
		});

		// Now create the box
	};

	return (
		<div className="NewBoxForm-container">
			<h2 className="NewBoxForm-title">New Box Form</h2>
			<form className="NewBoxForm" onSubmit={handleSubmit}>
				<label className="NewBoxForm-label">
					Width:
					<input
						className="NewBoxForm-input"
						type="text"
						name="width"
						value={newBoxFormData.width}
						onChange={handleChange}
						required
					/>
				</label>
				<br />
				<label className="NewBoxForm-label">
					Height:
					<input
						className="NewBoxForm-input"
						type="text"
						name="height"
						value={newBoxFormData.height}
						onChange={handleChange}
						required
					/>
				</label>
				<br />
				<label className="NewBoxForm-label">
					Background Color:
					<input
						className="NewBoxForm-input"
						type="text"
						name="backgroundColor"
						value={newBoxFormData.backgroundColor}
						onChange={handleChange}
						required
					/>
				</label>
				<br />

				<button className="NewBoxForm-btn" type="submit">
					Add New Box!
				</button>
			</form>
		</div>
	);
}

export default NewBoxForm;
