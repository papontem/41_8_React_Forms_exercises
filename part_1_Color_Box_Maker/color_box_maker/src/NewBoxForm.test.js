import { render } from "@testing-library/react";
import NewBoxForm from "./NewBoxForm";

test("renders NewBoxForm without crashing", () => {
	render(<NewBoxForm />);
});
