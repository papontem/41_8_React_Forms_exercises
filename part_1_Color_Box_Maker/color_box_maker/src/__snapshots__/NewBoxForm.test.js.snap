// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewBoxForm snapshot 1`] = `
<DocumentFragment>
  <div
    class="NewBoxForm-container"
  >
    <h2
      class="NewBoxForm-title"
    >
      New Box Form
    </h2>
    <form
      class="NewBoxForm"
    >
      <label
        class="NewBoxForm-label"
      >
        Width:
        <input
          class="NewBoxForm-input"
          name="width"
          required=""
          type="text"
          value=""
        />
      </label>
      <br />
      <label
        class="NewBoxForm-label"
      >
        Height:
        <input
          class="NewBoxForm-input"
          name="height"
          required=""
          type="text"
          value=""
        />
      </label>
      <br />
      <label
        class="NewBoxForm-label"
      >
        Background Color:
        <input
          class="NewBoxForm-input"
          name="backgroundColor"
          required=""
          type="text"
          value=""
        />
      </label>
      <br />
      <button
        class="NewBoxForm-btn"
        type="submit"
      >
        Add New Box!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
